forge-game/src/main/java/forge/game/ability/effects/AddPhaseEffect.java

```diff
@@ -39,36 +42,41 @@ public void resolve(SpellAbility sa) {
             afterPhase = currentPhase;
         }
 
-        // The original next phase following afterPhase
-        PhaseType nextPhase = PhaseType.getNext(afterPhase, isTopsy);
-        List<PhaseType> extraPhaseList = new ArrayList<>();
+        int num = sa.hasParam("NumPhases") ? AbilityUtils.calculateAmount(host, sa.getParam("NumPhases"), sa) : 1;
 
-        // Insert ExtraPhase
-        String extra = sa.getParam("ExtraPhase");
-        if (extra.equals("Beginning")) {
-            extraPhaseList.addAll(PhaseType.PHASE_GROUPS.get(0));
-        } else if (extra.equals("Combat")) {
-            extraPhaseList.addAll(PhaseType.PHASE_GROUPS.get(2));
-        } else { // Currently no effect will add End Phase
-            extraPhaseList.add(PhaseType.smartValueOf(extra));
-        }
+        while (num > 0) {
+            // The original next phase following afterPhase
+            PhaseType nextPhase = PhaseType.getNext(afterPhase, isTopsy);
+            List<PhaseType> extraPhaseList = new ArrayList<>();
 
-        // Insert FollowedBy
-        if (sa.hasParam("FollowedBy")) {
-            // Currently all FollowedBy are Main2 phase, which has no step
-            PhaseType followingExtra = PhaseType.smartValueOf(sa.getParam("FollowedBy"));
-            extraPhaseList.add(followingExtra);
-        }
+            // Insert ExtraPhase
+            String extra = sa.getParam("ExtraPhase");
+            if (extra.equals("Beginning")) {
+                extraPhaseList.addAll(PhaseType.PHASE_GROUPS.get(0));
+            } else if (extra.equals("Combat")) {
+                extraPhaseList.addAll(PhaseType.PHASE_GROUPS.get(2));
+            } else { // Currently no effect will add End Phase
+                extraPhaseList.add(PhaseType.smartValueOf(extra));
+            }
 
-        ExtraPhase extraPhase = phaseHandler.addExtraPhase(afterPhase, extraPhaseList, nextPhase);
+            // Insert FollowedBy
+            if (sa.hasParam("FollowedBy")) {
+                // Currently all FollowedBy are Main2 phase, which has no step
+                PhaseType followingExtra = PhaseType.smartValueOf(sa.getParam("FollowedBy"));
```

Comment:
We can pull the sa.getParam above the loop too. Then just check if followingExtra isn't null here instead.